test/over_react/component_declaration/builder_integration_tests/new_boilerplate/prop_conversion_annotations_test.dart

```diff
@@ -0,0 +1,83 @@
+// @dart=2.14
+// Copyright 2024 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:over_react/over_react.dart';
+import 'package:react_testing_library/react_testing_library.dart' as rtl;
+import 'package:test/test.dart';
+
+part 'prop_conversion_annotations_test.over_react.g.dart';
+
+void main() {
+  group('(New boilerplate) prop conversion annotations:', () {
```

Comment:
Not sure if this is the best way to test this functionality 🤷‍♀️

Reply:
Another way could be to verify behavior using raw values within the underlying map:

```dart
test('@ConvertProp works as expected', () {
  final fooPropKey = Test().getPropKey((p) => p.foo);

  expect((Test()..foo = 100), equals({fooPropKey: r'$100'}), 
      reason: 'setter should convert value');

  expect(Test({fooPropKey: r'$200'}).foo, 200, 
      reason: 'getter should convert value');
});
```

Reply:
ooo I love this! Thank you!!